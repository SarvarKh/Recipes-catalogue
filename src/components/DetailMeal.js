import { connect } from "react-redux";

const DetailMeal = ({detailMeal}) => {
    return (
        <main>
            {
                detailMeal !== undefined ?
                    <div>
                        <h1>{detailMeal.strMeal}</h1>
                        <div>Category: {detailMeal.strCategory}</div>
                        <div>Origin: {detailMeal.strArea}</div>
                        <div className="instructions">
                            <p>How to prepare?</p>
                            <p>{detailMeal.strInstructions}</p>
                        </div>
                        <div>
                            <img src={detailMeal.strMealThumb} />
                        </div>
                        <div>#Tags: {detailMeal.strTags}</div>
                        <iframe width="420" height="345" src={`https://www.youtube.com/embed/${detailMeal.strYoutube.split('=')[1]}`}></iframe>
                        
                        <table>
                            <tr>
                                <th>No</th>
                                <th>Ingredients</th>
                                <th>Measure</th>
                            </tr>
                            {detailMeal.strIngredient1 ?
                                <tr>
                                    <td>1</td>
                                    <td>{detailMeal.strIngredient1}</td>
                                    <td>{detailMeal.strMeasure1}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient2 ?
                                <tr>
                                    <td>2</td>
                                    <td>{detailMeal.strIngredient2}</td>
                                    <td>{detailMeal.strMeasure2}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient3 ?
                                <tr>
                                    <td>3</td>
                                    <td>{detailMeal.strIngredient3}</td>
                                    <td>{detailMeal.strMeasure3}</td>
                                </tr>
                                : ""}
                            
                                {detailMeal.strIngredient4 ?
                                <tr>
                                    <td>4</td>
                                    <td>{detailMeal.strIngredient4}</td>
                                    <td>{detailMeal.strMeasure4}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient5 ?
                                <tr>
                                    <td>5</td>
                                    <td>{detailMeal.strIngredient5}</td>
                                    <td>{detailMeal.strMeasure5}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient6 ?
                                <tr>
                                    <td>6</td>
                                    <td>{detailMeal.strIngredient6}</td>
                                    <td>{detailMeal.strMeasure6}</td>
                                </tr>
                                : ""}
                            
                            {detailMeal.strIngredient7 ?
                                <tr>
                                    <td>7</td>
                                    <td>{detailMeal.strIngredient7}</td>
                                    <td>{detailMeal.strMeasure7}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient8 ?
                                <tr>
                                    <td>8</td>
                                    <td>{detailMeal.strIngredient8}</td>
                                    <td>{detailMeal.strMeasure8}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient9 ?
                                <tr>
                                    <td>9</td>
                                    <td>{detailMeal.strIngredient9}</td>
                                    <td>{detailMeal.strMeasure9}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient10 ?
                                <tr>
                                    <td>10</td>
                                    <td>{detailMeal.strIngredient10}</td>
                                    <td>{detailMeal.strMeasure10}</td>
                                </tr>
                                : ""}

                            {detailMeal.strIngredient11 ?
                                <tr>
                                    <td>11</td>
                                    <td>{detailMeal.strIngredient1}</td>
                                    <td>{detailMeal.strMeasure1}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient12 ?
                                <tr>
                                    <td>12</td>
                                    <td>{detailMeal.strIngredient12}</td>
                                    <td>{detailMeal.strMeasure12}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient13 ?
                                <tr>
                                    <td>13</td>
                                    <td>{detailMeal.strIngredient13}</td>
                                    <td>{detailMeal.strMeasure13}</td>
                                </tr>
                                : ""}
                            
                                {detailMeal.strIngredient14 ?
                                <tr>
                                    <td>14</td>
                                    <td>{detailMeal.strIngredient14}</td>
                                    <td>{detailMeal.strMeasure14}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient15 ?
                                <tr>
                                    <td>15</td>
                                    <td>{detailMeal.strIngredient15}</td>
                                    <td>{detailMeal.strMeasure15}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient16 ?
                                <tr>
                                    <td>16</td>
                                    <td>{detailMeal.strIngredient16}</td>
                                    <td>{detailMeal.strMeasure16}</td>
                                </tr>
                                : ""}
                            
                            {detailMeal.strIngredient17 ?
                                <tr>
                                    <td>17</td>
                                    <td>{detailMeal.strIngredient17}</td>
                                    <td>{detailMeal.strMeasure17}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient18 ?
                                <tr>
                                    <td>18</td>
                                    <td>{detailMeal.strIngredient18}</td>
                                    <td>{detailMeal.strMeasure18}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient19 ?
                                <tr>
                                    <td>19</td>
                                    <td>{detailMeal.strIngredient19}</td>
                                    <td>{detailMeal.strMeasure19}</td>
                                </tr>
                                : ""}
                            {detailMeal.strIngredient20 ?
                                <tr>
                                    <td>20</td>
                                    <td>{detailMeal.strIngredient20}</td>
                                    <td>{detailMeal.strMeasure20}</td>
                                </tr>
                                : ""}
                            
                        </table>
                        <a src={detailMeal.strSource}>Source</a>
                    </div>
                    
                    : <h2>Please go back to the previous page and select meal to see details...</h2>
            }
        </main>
    )
}

const mapStateToProps = (state) => {
    return {
      detailMeal: state.detailMeal.detailMeal,
    }
  }

export default connect(mapStateToProps, null)(DetailMeal);