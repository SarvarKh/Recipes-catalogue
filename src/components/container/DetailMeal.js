import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useParams } from 'react-router-dom';
import MapDetailMeal from '../presentation/MapDetailMeal';
import { fetchDetailMeal } from '../../actions/index';

const DetailMeal = ({ detailMeal, fetchDetailMeal }) => {
  const { id } = useParams();
  useEffect(() => {
    fetchDetailMeal(id);
  }, []);
  return (
    <main>
      {
        Object.keys(detailMeal).length !== 0
          ? <MapDetailMeal detailMeal={detailMeal} />
          : (
            <h2>
              Please go back to the previous page and select meal to see details...
              {id}
            </h2>
          )
      }
    </main>
  );
};

const mapStateToProps = (state) => ({
  detailMeal: state.detailMeal.detailMeal,
});

export default connect(mapStateToProps, { fetchDetailMeal })(DetailMeal);

DetailMeal.propTypes = {
  detailMeal: PropTypes.instanceOf(Object).isRequired,
  fetchDetailMeal: PropTypes.func.isRequired,
};
