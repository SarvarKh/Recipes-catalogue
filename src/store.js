import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './reducers/rootReducer';

const initialState = {
  detailMeal: {
    detailMeal: {},
  },
  meals: {
    meals: [],
  },
  categories: {
    categories: [],
  },
  searchedItems: [],
};

const middleware = [thunk];

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(
    applyMiddleware(...middleware),
  ),
);

export default store;
