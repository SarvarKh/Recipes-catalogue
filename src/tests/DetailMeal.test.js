import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../store';
import DetailMeal from '../components/container/DetailMeal';

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><DetailMeal /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
