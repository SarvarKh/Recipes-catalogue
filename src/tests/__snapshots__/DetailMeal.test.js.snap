// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <h2>
    Please go back to the previous page and select meal to see details...
  </h2>
</main>
`;
